### PR TITLE
DB-12213 unit tests for code generation

### PR DESCRIPTION
**Pull request by martinrupp**
## Short Description
adding unit tests for code generation and some convenience methods.

## Long Description
this should make code generation a bit better accessible (test are more or less examples), faster development (can test some code generation in tests, don't need to start full DB for it), and makes refactoring more safe.

## How to test
No external testing necessary, no functional code changes.

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3126/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/test/java/com/splicemachine/derby/utils/CodeGenUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+        }
+        String path = "/tmp/";
+        String filename = path + cb.getName() + ".class";
+        try {
+            cb.writeClassFile(path, false, null);
+            String output = execSysCommand("javap " + options + " " + filename);
+            output = output.replace(cb.getFullName(), "%CLASSNAME%");
+            Assert.assertEquals(expected, output);
+        } finally {
+            //new File(filename).delete();
```
Comment (martinrupp) [inline splice_machine/src/test/java/com/splicemachine/derby/utils/CodeGenUnitTest.java:484]: please re-add

Comment (martinrupp) [reply splice_machine/src/test/java/com/splicemachine/derby/utils/CodeGenUnitTest.java]: done

**Review by arnaud-splice — APPROVED** [review]

**Review by jyuanca — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**